build-logic/build.gradle.kts

```diff
@@ -17,6 +24,39 @@ kotlin {
             .get()
             .toInt(),
     )
+
+    compilerOptions {
+        // Match language version enforced by current Gradle version https://docs.gradle.org/current/userguide/compatibility.html#kotlin
+        @Suppress("DEPRECATION")
+        apiVersion.set(KotlinVersion.KOTLIN_1_8)
+    }
+}
+
+tasks.withType<KotlinCompile>().configureEach {
+    compilerOptions {
+        freeCompilerArgs.add("-Xsuppress-version-warnings") // ignores deprecated kotlin language version
+    }
+}
+
+gradlePlugin {
+    plugins {
+        register("ktlint-dokka") {
+            id = "ktlint-dokka"
+            implementationClass = "DokkaPlugin"
+        }
+        register("ktlint-kotlin-common") {
+            id = "ktlint-kotlin-common"
+            implementationClass = "KotlinCommonPlugin"
+        }
+        register("ktlint-publication") {
+            id = "ktlint-publication"
+            implementationClass = "PublicationPlugin"
+        }
+        register("ktlint-publication-library") {
+            id = "ktlint-publication-library"
+            implementationClass = "PublicationLibraryPlugin"
+        }
+    }
```

Comment:
This is the overhead binary plugins add. Previously the name of the plugin was driven by the file name. Now the bindings are defined here